synapseclient/models/submissionview.py

```diff
@@ -0,0 +1,923 @@
+    in Synapse with the new name and the key in the OrderedDict will be updated.
+    """
+
+    _columns_to_delete: Optional[Dict[str, Column]] = field(default_factory=dict)
+    """
+    Columns to delete when the submission view is stored. The key in this dict is the ID of the
+    column to delete. The value is the Column object that represents the column to
+    delete.
+    """
+
+    activity: Optional[Activity] = field(default=None, compare=False)
+    """The Activity model represents the main record of Provenance in Synapse.  It is
+    analogous to the Activity defined in the
+    [W3C Specification](https://www.w3.org/TR/prov-n/) on Provenance."""
+
+    annotations: Optional[
+        Dict[
+            str,
+            Union[
+                List[str],
+                List[bool],
+                List[float],
+                List[int],
+                List[date],
+                List[datetime],
+            ],
+        ]
+    ] = field(default_factory=dict, compare=False)
+    """Additional metadata associated with the submission view. The key is the name of your
+    desired annotations. The value is an object containing a list of values
+    (use empty list to represent no values for key) and the value type associated with
+    all values in the list. To remove all annotations set this to an empty dict `{}`"""
+
+    _last_persistent_instance: Optional["SubmissionView"] = field(
+        default=None, repr=False, compare=False
+    )
+    """The last persistent instance of this object. This is used to determine if the
+    object has been changed and needs to be updated in Synapse."""
+
+    is_search_enabled: Optional[bool] = None
+    """
+    When creating or updating a table or view, specifies if full-text search should be enabled.
+    Note that enabling full-text search might slow down the indexing of the table or view.
+    """
+
+    scope_ids: List[str] = field(default_factory=list)
+    """
+    The list of container IDs that define the scope of this view. For submission views,
+    this is the list of evaluation queues that the view is associated with.
+    """
+
+    view_entity_type: ViewEntityType = ViewEntityType.SUBMISSION_VIEW
+    """The API model string for the type of view. This is used to determine the default columns that are
+    added to the table. Must be defined as a `ViewEntityType` enum.
+    """
+
+    def __post_init__(self):
+        self.columns = self._convert_columns_to_ordered_dict(columns=self.columns)
+
+    @property
+    def has_changed(self) -> bool:
+        """Determines if the object has been changed and needs to be updated in Synapse."""
+        return (
+            not self._last_persistent_instance
+            or self._last_persistent_instance != self
+            or (not self._last_persistent_instance.scope_ids and self.scope_ids)
+            or self._last_persistent_instance.scope_ids != self.scope_ids
+        )
+
+    @property
+    def has_columns_changed(self) -> bool:
```

Comment:
Just for my understanding, is this not part of the list of classes that this is composed of already?

Reply:
Thats right, however, I moved this into the ColumnMixin and removed this code from 5 places it was duplicated.